lib/models/build.coffee

```diff
@@ -40,23 +40,25 @@ getBuildModel = (deps, opts) ->
 	# });
 	#
 	# @example
+	# resin.models.build.getAllByApplication(123).then(function(builds) {
+	#		console.log(builds);
+	# });
+	#
+	# @example
 	# resin.models.build.getAllByApplication('MyApp', function(error, builds) {
 	#		if (error) throw error;
 	#		console.log(builds);
 	# });
 	###
-	exports.getAllByApplication = (name, callback) ->
-		applicationModel().has(name).then (hasApplication) ->
-
-			if not hasApplication
-				throw new errors.ResinApplicationNotFound(name)
+	exports.getAllByApplication = (nameOrId, callback) ->
+		applicationModel().get(nameOrId).then (application) ->
 
 			return pine.get
 				resource: 'build'
 				filter:
 					application: $any:
```

Comment:
I think it you can change this to `application: application.id` now that it's using id